lib/coney/connection_server.ex

```diff
@@ -40,28 +56,62 @@ defmodule Coney.ConnectionServer do
     GenServer.call(__MODULE__, {:publish, exchange_name, routing_key, message})
   end
 
+  def subscribe(consumer) do
+    GenServer.call(__MODULE__, {:subscribe, consumer})
+  end
+
+  @impl GenServer
   def handle_info(:after_init, state) do
-    rabbitmq_connect(state)
+    {:noreply, rabbitmq_connect(state)}
   end
 
   def handle_info({:DOWN, _, :process, _pid, reason}, state) do
     ConnectionRegistry.disconnected(self())
     Logger.error("#{__MODULE__} (#{inspect(self())}) connection lost: #{inspect(reason)}")
-    rabbitmq_connect(state)
+    {:noreply, state |> rabbitmq_connect() |> update_channels()}
   end
 
-  def terminate(_reason, %State{amqp_conn: conn, adapter: adapter} = _state) do
+  @impl GenServer
+  def terminate(_reason, %State{amqp_conn: conn, adapter: adapter, channels: channels} = _state) do
+    close_channels(channels, adapter)
```

Comment:
Maybe log here, Instead of every channel close?

Reply:
Notifying on each channel close gives us more information on the number of workers, how long it takes to close each channel, etc.

Reply:
Well, the application can have hundreds of channels open and we don't want to flood the logs. Modified to log a single message with the number of channels closed